dev/doc/changes.md

```diff
@@ -18,6 +18,12 @@ General deprecation
   removed. Please, make sure your plugin is warning-free in 8.8 before
   trying to port it over 8.9.
 
+Warnings
+
+- Coq now build plugins with `-warn-error` by default, the amount of
+  dangerous warnings in plugin code was very high, we now thus require
+  plugins in the CI to be warning-free to be included.
```

Comment:
You should probably add a line to explain how to turn off `-warn-error`.

Reply:
That should never be done IMHO, but individual `[@ocaml.warning ...]` declarations should be added to the code.

Reply:
In particular I want to discourage plugins authors from modifying ocamlflags at all. If they have such special needs they should know how to do it.

Reply:
Then please add this:

>That should never be done IMHO, but individual `[@ocaml.warning ...]` declarations should be added to the code.

Reply:
I don't want to encourage this sorry; it'd look like we recommend doing that and it is strongly not recommended to do so, but to fix the warnings instead. There is no good reason to disable any of the warnings we enable. Plugins authors are competent ocaml devs so if they have a reason they'll know what to do.

Reply:
>Plugins authors are competent ocaml devs

You are making wrong assumptions on the background of plugin developers. They are Coq users who happen to need to program in OCaml to achieve what they want to do.

But OK, if you think that this is not going to make the life of plugin developers so much harder, I'm fine with not adding anything.

Reply:
Why is my assumption wrong? I know most of the authors personally and indeed they are very competent devs. In fact, I am skeptical you can get a functioning coq plug-in unless you are very competent in ocaml.

Reply:
Indeed your feedback @Zimmi48 has made me think to reword the message even more strongly; there is no reason to ever disable any of the warnings we enable in "production" code so I'll make the policy and wording more strict.